chunky/src/java/se/llbit/chunky/renderer/scene/PreviewRayTracer.java

```diff
@@ -83,21 +85,27 @@ public static double skyOcclusion(Scene scene, WorkerState state) {
 
   /**
    * Find next ray intersection.
+   * @param random Used for particle fog, can be null if particleFog is false
    * @return Next intersection
    */
-  public static boolean nextIntersection(Scene scene, Ray ray) {
+  public static boolean nextIntersection(Scene scene, Ray ray, Random random, IntersectionConfig config) {
```

Comment:
I think it would be better for this to give the intersection with the entire fog volume instead of needing to pass in a random and intersect in here. That would give ray tracers more flexibility on what it wants to do with the fog (ie ignore it, intersect with fog, calculate attenuation of fog, etc).